### PR TITLE
Eventhubs remove iot and linkredirect

### PR DESCRIPTION
**Pull request by YijunXieMS**
Remove iot hub and link redirect related code

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/eventhub/azure-eventhubs/azure/eventhub/client.py [hunk 1]
```diff
@@ -261,19 +238,17 @@ def create_consumer(self, consumer_group, partition_id, event_position, **kwargs
...
+            self._address.hostname, self._address.path, consumer_group, partition_id)
         handler = EventHubConsumer(
             self, source_url, event_position=event_position, owner_level=owner_level,
             prefetch=prefetch)
         return handler
 
-    def create_producer(self, partition_id=None, operation=None, send_timeout=None):
-        # type: (str, str, float) -> EventHubProducer
+    def create_producer(self, partition_id=None, send_timeout=None):
+        # type: (str, str) -> EventHubProducer
```
Comment (yunhaoling) [inline sdk/eventhub/azure-eventhubs/azure/eventhub/client.py]: `# type: (str, float) -> EventHubProducer`

File: sdk/eventhub/azure-eventhubs/azure/eventhub/aio/consumer_async.py [hunk 2]
```diff
@@ -199,7 +172,7 @@ def queue_size(self):
         return 0
 
     async def receive(self, *, max_batch_size=None, timeout=None):
-        # type: (int, float) -> List[EventData]
+        # type: (Any, int, float) -> List[EventData]
```
Comment (yunhaoling) [inline sdk/eventhub/azure-eventhubs/azure/eventhub/aio/consumer_async.py]: Do we need this 'Any' in the type description? as the `*` is just for omitting positional args..we simply ignore position args.

Comment (YijunXieMS) [reply sdk/eventhub/azure-eventhubs/azure/eventhub/aio/consumer_async.py]: Without "Any" Pycharm would report type hint error and have funny code assistant.

File: sdk/eventhub/azure-eventhubs/azure/eventhub/client.py (hunk 1 above)
Comment (YijunXieMS) [reply sdk/eventhub/azure-eventhubs/azure/eventhub/client.py]: good catch. fixed

File: sdk/eventhub/azure-eventhubs/azure/eventhub/client_abstract.py [hunk 3]
```diff
@@ -196,40 +162,30 @@ def _create_properties(self, user_agent=None):  # pylint: disable=no-self-use
...
         """
         properties = {}
-        properties["product"] = "eventhub.python"
-        properties["version"] = __version__
-        properties["framework"] = "Python {}.{}.{}".format(*sys.version_info[0:3])
-        properties["platform"] = sys.platform
-
-        final_user_agent = 'azsdk-python-eventhub/{} ({}; {})'.format(
-            __version__, properties["framework"], sys.platform)
+        product = "azure-eventhub"
```
Comment (yunhaoling) [inline sdk/eventhub/azure-eventhubs/azure/eventhub/client_abstract.py]: product = "azsdk-python-eventhubs"

**Pull request merged**